File: MedArrival-Backend/src/main/java/com/cnesten/medarrivalbackend/DTO/SaleDTO.java
package com.cnesten.medarrivalbackend.DTO;

import lombok.Data;

import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

@Data
public class SaleDTO {
    private Long id;
    private Integer quantity;
    private Integer expectedQuantity;
    private Float totalAmount;
    private LocalDateTime saleDate;
    private LocalDateTime expectedDeliveryDate;
    private Boolean isConform;
    private ProductDTO product;
    private Set<SalePriceComponentDTO> priceComponents = new HashSet<>();
    private ClientDTO client;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String createdBy;
    private String updatedBy;
}

File: MedArrival-Front/src/models/SaleDTO.ts
import { BaseDTO } from "./BaseDTO";
import { ClientDTO } from "./ClientDTO";
import { ProductDTO } from "./ProductDTO";

export enum PriceComponentType {
    PURCHASE_PRICE = 'PURCHASE_PRICE',
    TRANSPORT = 'TRANSPORT',
    STORAGE = 'STORAGE',
    TRANSIT = 'TRANSIT',
    DUANE = 'DUANE',
    AMSSNUR = 'AMSSNUR'
}

export interface SalePriceComponentDTO extends BaseDTO {
    componentType: PriceComponentType;
    amount: number;
    usesDefaultPrice: boolean;
}

export interface SaleDTO extends BaseDTO {
    product: ProductDTO;
    client: ClientDTO;
    quantity: number;
    priceComponents: SalePriceComponentDTO[];
    expectedDeliveryDate: Date;
    isConform: boolean;
    totalAmount: number;
    expectedQuantity: number;
    saleDate: Date;
}